### PR TITLE
ci: Use macos-latest

### PR DESCRIPTION
**Pull request by abetomo**
The issue has been fixed.
https://github.com/Homebrew/homebrew-core/issues/171137

### CODE REVIEW

**Comment by abetomo** [comment]
MeCab issue has been fixed and we have reverted back to `macos-latest`, but the test seems to fail due to other issues.
We will check the contents and fix them.

Related: #436

**Comment by kou** [comment]
https://github.com/pgroonga/pgroonga/actions/runs/9375615136/job/25814034986?pr=483#step:10:9259

```text
2024-06-04 23:30:09.177275|e|24858: system call error: Too many open files: [io][open] failed to open path: <base/16384/pgrn.00006B6>
```

This is the problem.

**Comment by askdkc** [comment]
Ah, too many open files!! The most painful part while using PGroonga on macOS!!
It requires changing the system setting like this:

```bash
sudo vi /Library/LaunchDaemons/limit.maxfiles.plist
```

## these parameter works most of the time
```vim
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">  
  <dict>
    <key>Label</key>
    <string>limit.maxfiles</string>
    <key>ProgramArguments</key>
    <array>
      <string>launchctl</string>
      <string>limit</string>
      <string>maxfiles</string>
      <string>524288</string>
      <string>2048576</string>
    </array>
    <key>RunAtLoad</key>
    <true/>
    <key>ServiceIPC</key>
    <false/>
    <key>HardResourceLimits</key>
    <dict>
      <key>NumberOfFiles</key>
      <integer>64000</integer>
    </dict>
    <key>SoftResourceLimits</key>
    <dict>
      <key>NumberOfFiles</key>
      <integer>64000</integer>
    </dict>
  </dict>
</plist>
```

## Apply above using following command
```bash
sudo launchctl load -w /Library/LaunchDaemons/limit.maxfiles.plist 
```

**Comment by abetomo** [comment]
@kou @askdkc Thanks for the information.
I was able to set it up with the following command.

```
sudo launchctl limit maxfiles unlimited unlimited
```

**Comment by abetomo** [comment]
The maxfiles issue has been resolved, but the test is failing due to other issue, so I will check it.
https://github.com/pgroonga/pgroonga/actions/runs/9983174649/job/27590276340#step:11:95

```
Using libpq from 
checking for libpq-fe.h... no
Can't find the 'libpq-fe.h header
```

**Comment by kou** [comment]
This may solve it:

```diff
diff --git a/.github/workflows/macos.yml b/.github/workflows/macos.yml
index 3095fbed..49b86e14 100644
--- a/.github/workflows/macos.yml
+++ b/.github/workflows/macos.yml
@@ -74,6 +74,8 @@ jobs:
           random_page_cost = 0
           CONF
           brew services start postgresql@${{ matrix.postgresql-version }}
+          echo $(brew --prefix postgresql@${{ matrix.postgresql-version }})/bin >> \
+            ${GITHUB_PATH}
       - name: Run regression test
         run: |
           if [ ${{ matrix.postgresql-version }} -lt 13 ]; then
```

File: .github/workflows/macos.yml [hunk 1]
```diff
@@ -66,6 +66,9 @@ jobs:
             -e 's/postgresql/postgresql@${{ matrix.postgresql-version }}/g' \
             Brewfile
           brew bundle
+      - name: Update maxfiles limit
+        # To avoid the error `system call error: Too many open files`.
+        run: sudo launchctl limit maxfiles unlimited unlimited
```
Comment (askdkc) [inline .github/workflows/macos.yml:71]: This is golden!!

**Review by komainu8 — APPROVED** [review]

**Pull request merged**